gate-proxy/src/main/kotlin/com/netflix/spinnaker/gate/controllers/Proxy.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.controllers
+
+import com.netflix.spinnaker.gate.api.extension.ProxyConfig
+import com.squareup.okhttp.OkHttpClient
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.security.KeyStore
+import java.security.cert.X509Certificate
+import java.util.concurrent.TimeUnit
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import javax.net.ssl.TrustManagerFactory
+import javax.net.ssl.X509TrustManager
+
+internal class Proxy(val config: ProxyConfig) {
+  companion object {
+    val logger = LoggerFactory.getLogger(ProxyConfig::class.java)
+  }
+
+  var okHttpClient = OkHttpClient()
+
+  fun init() {
```

Comment:
Not for this PR, but I think it would set us up nicely if we provided these proxy clients from the kork `OkHttpClientProvider` eventually.

Reply:
Ya, that's a fair point to revisit.